### PR TITLE
Google Analytics - IP Anonymization

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5483
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. ) 

[//]: # ( Required: )
#### Description:
Just added new option to  anonymize the IP address for some country law restrictions. More info https://developers.google.com/analytics/devguides/collection/analyticsjs/ip-anonymization

image

#### Steps to test this PR:
1.  Apply PR and setup Google Analytics in Configuration > Tracking Settings tab
2.  Enable IP Anonymization and Enable on Mautic landing page
3. Go to Mautic landing page and see source if  this line was added
`ga('set', 'anonymizeIp', true);`

### CODE REVIEW

**Comment by kuzmany** [comment]
Label: Ready to test

File: app/bundles/PageBundle/EventListener/BuildJsSubscriber.php [hunk 1]
```diff
@@ -547,7 +550,8 @@ public function onBuildJsForTrackingEvent(BuildJsEvent $event)
   })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
   ga('create', '{$id}', 'auto');
   ga('send', 'pageview');
-  ga('set', 'userId', {$leadId});
+  {$gaUserId}
+  {$gaAnonymizeIp}
```
Comment (jimspillane) [inline app/bundles/PageBundle/EventListener/BuildJsSubscriber.php]: The IP anonymization parameter looks like this: _&aip=1_  It is not being sent to GA on the a hit of the LP.  Move `{$gaAnonymizeIp}`  before `ga('send', 'pageview');` so the parameter is added.

Comment (kuzmany) [reply app/bundles/PageBundle/EventListener/BuildJsSubscriber.php]: You're right. Moved set before event. Thanks

**Comment by escopecz** [comment]
@kuzmany is adding `plugins/MauticMailTesterBundle` submodule intentional?

File: app/bundles/PageBundle/Form/Type/ConfigTrackingPageType.php [hunk 2]
```diff
@@ -112,6 +112,18 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'data'  => (bool) $options['data']['google_analytics_landingpage_enabled'],
             ]
         );
+
+        $builder->add(
+            'google_analytics_anonymize_ip',
+            'yesno_button_group',
+            [
+                'label' => 'mautic.page.config.form.tracking.anonymize.ip.enabled',
+                'data'  => (bool) $options['data']['google_analytics_anonymize_ip'],
```
Comment (escopecz) [inline app/bundles/PageBundle/Form/Type/ConfigTrackingPageType.php]: I'm getting `PHP Notice - Undefined index: google_analytics_anonymize_ip` on this line when I apply this PR, clear the cache and access the Mautic configuration.

Comment (kuzmany) [reply app/bundles/PageBundle/Form/Type/ConfigTrackingPageType.php]: Okay, issue depends on your fix https://github.com/mautic/mautic/pull/5663
I added param to ConfigSubsriber. Now should works

Comment (escopecz) [reply app/bundles/PageBundle/Form/Type/ConfigTrackingPageType.php]: That fix is merged already so it should be included in your PR too. The problem is that the $options['data'] doesn't contain the new property yet.

Comment (kuzmany) [reply app/bundles/PageBundle/Form/Type/ConfigTrackingPageType.php]: But this line fixed it, didn't it? https://github.com/mautic/mautic/pull/5525/commits/d763b0a14ca4f2129dd7b9313455010761111147#diff-2ede0d259417f4d170e6c0f8b2cbf442R71

Comment (escopecz) [reply app/bundles/PageBundle/Form/Type/ConfigTrackingPageType.php]: I don't think so. Try to remove that value from your app/config/local.php, clear the cache and try to load the configuration.

**Comment by kuzmany** [comment]
Label: WIP

**Comment by kuzmany** [comment]
Label: Pending feedback

**Comment by npracht** [comment]
@kuzmany to make it GDPR compliant, we'd need to:
1. hash the IPs in DB
2. make sure the IP is not fully visible for user on the interface

Could you please work on it ?

**Comment by kuzmany** [comment]
@npracht of course, but maybe somebody from Mautic core team already work on it (I heart about it). @alanhartless  @dbhurley  ?

**Comment by npracht** [comment]
This PR is ready. The option to anonimyze IP in the interface is managed by another PR.

**Comment by heathdutton** [comment]
Feels like the tooltip is sufficient documentation IMO. "In some cases, you might need to anonymize the IP address of the hit sent to Google Analytics".

**Pull request merged**